packages/odyssey-react/src/components/Tooltip/index.tsx

```diff
@@ -52,12 +53,17 @@ const Tooltip: FunctionComponent<Props> = (props) => {
   const clone = cloneElement(children, { 'aria-describedby': oid });
 
   const tooltipClasses = useCx(
-    'ods-tooltip',
-    `is-ods-tooltip-${position}`
+    styles.root,
+    {
+      [styles.positionTop]: position === 'top',
```

Comment:
I've opted for additional lines instead of a splice here. I'm sensitive to the additional lines, but would prefer to find a way to keep this clarity in our class names.

Reply:
same feedback as before, consider the total number of lines of code here to apply what will only ever be one class (enforced through types and component API). The implication of code like this is that its possible to have many `position*` classes applied but that isn't the case. Consider a refactor like this:

```
styles.root,
styles[`${position}Position`]
```

rename your classes and be done!

Reply:
Naming our classes to always match prop values will be troublesome at scale. For authors, classes like `.s` or `.sSize` are hard to parse. For QA or other troubleshooters, rendered class names ideally contain enough info to understand why they're being applied, since the browser is the first point of contact, not the source.

I'm open to rearranging our class naming to be `.valueType`, but it's somewhat harder to parse and won't match our token naming conventions. 

Consider:

```
$size-small: 
$size-medium: 
$size-large:

$small-size:
$medium-size:
$large-size:

.sizeS
.sizeM
.sizeL

.sSize
.mSize
.lSize
```

I'm sensitive to code cruft and elegance, but we should maintain QoL benefits across the board, not just for React. If there's a solution that improves both sides, I'd like to try and get there first.

Reply:
@edburyenegren-okta thanks for the writeup.

I understand the motivation to have readable class names. I'm not suggesting we use this approach everywhere without thought or consideration. The class names for CSS Modules are malleable and the local class name is just one part of the final identifier that appears in storybook in development. It's likely it won't even appear in production builds. Optimizing for these types of concerns over the component code clarity (and even a tiny bit of performance *for every render* in this case) isn't the correct tradeoff in this case.

I don't believe that we need to spend time adopting broad standards across all components right now for these types of questions. CSS modules are encapsulated meaning we can reason about them one at a time. I've already added a lint rule to enforce the structure of class names. This feels like enough for now and I don't want our team focused on re-arranging class names that won't appear as written at develop time anyways.

Looking forward, I don't see a clear way to enforce `sizeS` over `sSize` via static analysis. Regardless, that this isn't a problem we need to solve for this PR.